.github/workflows/preview.yml

```
@@ -3,43 +3,91 @@ name: Publish Preview release
+            platform: win32
+            arch: x64
+            npm_config_arch: x64
+            target: win32-x64
+          - os: windows-latest
+            platform: win32
+            arch: ia32
+            npm_config_arch: ia32
+            target: win32-ia32
+          # - os: windows-latest
+          #   platform: win32
+          #   arch: arm64
+          #   npm_config_arch: arm
+          #   target: win32-arm64
+          - os: ubuntu-latest
+            platform: linux
+            arch: x64
+            npm_config_arch: x64
+            target: linux-x64
+          - os: ubuntu-latest
+            platform: linux
+            arch: arm64
+            npm_config_arch: arm64
+            target: linux-arm64
+          - os: ubuntu-latest
+            platform: linux
+            arch: armhf
+            npm_config_arch: arm
+            target: linux-armhf
+          - os: macos-latest
+            platform: darwin
+            arch: x64
+            npm_config_arch: x64
+            target: darwin-x64
+          - os: macos-latest
+            platform: darwin
+            arch: arm64
+            npm_config_arch: arm64
+            target: darwin-arm64
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
-
-      - name: Set preview in package.json
-        id: set-preview
-        run: |
-          ./build/preview
-
-      - name: Read extension version
-        id: ext-version
-        run: |
-          content=`cat ./package.json | jq -r .version`
-          echo "::set-output name=content::$content"
-
       - uses: actions/setup-node@v2
         with:
-          node-version-file: '.nvmrc'
-
+          node-version-file: ".nvmrc"
       - name: Install dependencies
         run: npm ci
-
-      - name: vsce package
-        run: npx -- vsce package --out=${{ runner.temp }}/terraform-${{ steps.ext-version.outputs.content }}.vsix
-
+        env:
+          npm_config_arch: ${{ matrix.npm_config_arch }}
+      - name: Set preview in package.json
+        id: set-preview
+        run: npm run preview
+      - name: Package VSIX
+        run: npm run package -- --target=${{ matrix.target }}
+        env:
+          target: ${{ matrix.target }}
+      - name: Upload vsix as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.target }}
+          path: "*.vsix"
       - name: Check latest published version
+        shell: bash
         run: |
           export EXTENSION_ID=`cat package.json | jq -r '.publisher + "." + .name'`
           npx -- vsce show --json $EXTENSION_ID | jq '.versions[0]'
 
-      - name: vsce publish
-        run: npx -- vsce publish --no-git-tag-version --packagePath=${{ runner.temp }}/terraform-${{ steps.ext-version.outputs.content }}.vsix
+  publish:
+    name: Publish All
+    runs-on: ubuntu-latest
+    needs: build
+    if: success() && startsWith( github.ref, 'refs/tags/')
```

```suggestion
    if: success() && startsWith( github.ref, 'refs/tags/v')
```
Totally nicpicky suggestion - I guess it won't make a difference but I'm thinking it would more accurately reflect intentions - of publishing on _version_ tags.